pandas/_libs/index.pyx

```diff
@@ -658,19 +740,54 @@ cdef class BaseMultiIndexCodesEngine:
         ----------
         target : list-like of keys
             Each key is a tuple, with a label for each level of the index.
+        method : string (optional)
+            whether to fill missing keys with either the previous (using "pad"
+            of "ffill" or next ("bfill"/"backfill") values, in terms of the
+            ordering of the tuples and the underlying index
 
         Returns
         ------
         int_keys : 1-dimensional array of dtype uint64 or object
             Integers representing one combination each
         """
-        level_codes = [lev.get_indexer(codes) + 1 for lev, codes
-                       in zip(self.levels, zip(*target))]
-        return self._codes_to_ints(np.array(level_codes, dtype='uint64').T)
+        level_codes = np.array([
+            lev.get_indexer(codes, method=method) for lev, codes
+            in zip(self.levels, zip(*target))
+        ], dtype='uint64').T + 1
```

Comment:
can any of this go directly in MultiIndex and not in the cython code?  the for-loops clearly belong here, but to the extent we can keep Index objects out of this, we should

Reply:
the updated version does not add any new uses of Index objects in the cython code